llm-examples/falcon/falcon-7b-instruct-inference.py

```diff
@@ -0,0 +1,349 @@
+    if use_template:
+        full_prompts = [
+            PROMPT_FOR_GENERATION_FORMAT.format(instruction=prompt)
+            for prompt in prompts
+        ]
+    else:
+        full_prompts = prompts
+
+    # configure text generation kwargs
+    if "max_new_tokens" not in kwargs:
+        kwargs["max_new_tokens"] = 512
+
+    if "batch_size" not in kwargs:
+        kwargs["batch_size"] = 8
+
+    kwargs.update(
+        {
+            "do_sample": True,
+            "use_cache": True,
+            "num_return_sequences": 1,
+            "pad_token_id": tokenizer.eos_token_id,
+            "eos_token_id": tokenizer.eos_token_id,
+        }
+    )
+
+    outputs = pipeline(full_prompts, **kwargs)
+    outputs = [out[0]["generated_text"] for out in outputs]
+
+    return outputs
+
+# COMMAND ----------
+
+# From databricks-dolly-15k
+inputs = [
+  "Think of some family rules to promote a healthy family relationship",
+  "In the series A Song of Ice and Fire, who is the founder of House Karstark?",
+  "which weighs more, cold or hot water?",
+  "Write a short paragraph about why you should not have both a pet cat and a pet bird.",
+  "Is beauty objective or subjective?",
+  "What is SVM?",
+  "What is the current capital of Japan?",
+  "Name 10 colors",
+  "How should I invest my money?",
+  "What are some ways to improve the value of your home?",
+  "What does fasting mean?",
+  "What is cloud computing in simple terms?",
+  "What is the meaning of life?",
+  "What is Linux?",
+  "Why do people like gardening?",
+  "What makes for a good photograph?"
+]
+
+# COMMAND ----------
+
+results = batch_gen_text(inputs, use_template=True)
+
+for output in results:
+  print(output)
+  print('\n')
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Use MLflow to manage the model
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Log the model
+
+# COMMAND ----------
+
+from huggingface_hub import snapshot_download
+
+# If the model has been downloaded in previous cells, this will not repetitively download large model files, but only the remaining files in the repo
+snapshot_location = snapshot_download(repo_id="tiiuae/falcon-7b-instruct",  ignore_patterns="coreml/*")
+
+# COMMAND ----------
+
+import mlflow
+
+# Define PythonModel to log with mlflow.pyfunc.log_model
+
+class Falcon(mlflow.pyfunc.PythonModel):
+    def load_context(self, context):
+        """
+        This method initializes the tokenizer and language model
+        using the specified model repository.
+        """
+        # Initialize tokenizer and language model
+        self.tokenizer = transformers.AutoTokenizer.from_pretrained(
+            context.artifacts['repository'], padding_side="left")
+        self.model = transformers.AutoModelForCausalLM.from_pretrained(
+            context.artifacts['repository'], 
+            torch_dtype=torch.bfloat16,
+            low_cpu_mem_usage=True, 
+            trust_remote_code=True,
```

Comment:
I remember you also need to use specified the "hash" to pin down the model somewhere, right?

Reply:
Yes, added `revision=...` in `snapshot_download`.